sdk/analytics/src/main/java/com/klaviyo/analytics/networking/KlaviyoApiClient.kt

```diff
@@ -170,22 +169,29 @@ internal object KlaviyoApiClient : ApiClient {
     }
 
     /**
-     * Enqueues an [KlaviyoApiRequest] to run in the background
+     * Enqueues one or more [KlaviyoApiRequest]s to send on a background thread
      * These requests are sent to the Klaviyo asynchronous APIs
      *
      * This method will initialize the API queue and the batching thread
      * if this is the first request made since launch.
      */
-    fun enqueueRequest(vararg requests: KlaviyoApiRequest) {
+    fun enqueueRequest(vararg requests: KlaviyoApiRequest, headOfLine: Boolean = false) {
```

Comment:
@ajaysubra I thought you'd be curious to see this update, prompted by your suggestion on an earlier PR (this is the same code, I just broke it up into 2 PRs for ease of review). 

This argument allows the priority events like opened push and geofence events to jump the queue, so in addition to flushing immediately, we'll move it to the front of the line (if there is a line anyway)

Reply:
Nice. @belleklaviyo FYI.